packages/react-router/index.tsx

```diff
@@ -579,9 +579,12 @@ export function useOutletContext<Context = unknown>(): Context {
  */
 export function useOutlet(context?: unknown): React.ReactElement | null {
   let outlet = React.useContext(RouteContext).outlet;
-  return (
-    <OutletContext.Provider value={context}>{outlet}</OutletContext.Provider>
-  );
+  if (context) {
```

Comment:
I would probably use `if (outlet)` here, because `useOutlet({ some: 'context' })` without a child route should still return `null`. Does that make sense @mcansh?